static/js/pages/applications/ApplicationDashboardPage.js

```diff
@@ -69,23 +114,168 @@ const getAppStepSubmissions = (
   )
 
 type Props = {
+  location: Location,
   applications: Array<Application>,
   allApplicationDetail: ApplicationDetailState,
+  appDetailQueryStatus: (applicationId: string) => ?QueryState,
   currentUser: User,
-  fetchAppDetail: (applicationId: number) => void,
-  openDrawer: (actionPayload: DrawerChangePayload) => void
+  fetchAppDetail: (
+    applicationId: string,
+    force?: boolean
+  ) => ?ApplicationDetailResponse,
+  fetchOrder: (orderId: string) => OrderResponse,
+  openDrawer: (actionPayload: DrawerChangePayload) => void,
+  addSuccessNotification: (actionPayload: any) => void,
+  addErrorNotification: (actionPayload: any) => void
 }
 
 type State = {
-  collapseVisible: Object
+  collapseVisible: { string: boolean },
+  pollingCount: number
 }
 
 export class ApplicationDashboardPage extends React.Component<Props, State> {
   state = {
-    collapseVisible: {}
+    collapseVisible: {},
+    pollingCount:    0
+  }
+
+  async componentDidMount() {
+    const { applications } = this.props
+    if (applications) {
+      const orderId = this.getOrderIdFromCybersourceParams()
+      if (orderId) {
+        await this.handleCybersourcePageLoad(orderId)
+      }
+    }
+  }
+
+  async componentDidUpdate(prevProps: Props, prevState: State) {
+    const { pollingCount } = this.state
+    if (
+      prevProps.applications !== this.props.applications ||
+      prevState.pollingCount !== pollingCount
+    ) {
+      const orderId = this.getOrderIdFromCybersourceParams()
+      if (orderId) {
+        if (pollingCount === 0) {
+          await this.handleCybersourcePageLoad(orderId)
+        } else if (prevState.pollingCount !== pollingCount) {
+          await this.checkForOrderCompletion(orderId)
+        }
+      }
+    }
   }
 
-  onCollapseToggle = (applicationId: number): void => {
+  getOrderIdFromCybersourceParams = (): ?string => {
+    const { location } = this.props
```

Comment:
Lots of new order-related code in this PR, is there another issue besides the two mentioned in the PR that it addresses?

Reply:
In a sense, yes. The order-related code handles the case where a user is returning from Cybersource and we want to show them a message if their payment succeeded or failed. We never ended up creating a discrete issue for that, though we should have. I just took care of it in this PR